### PR TITLE
API Fetch: Make preloaded OPTIONS requests use `parse` setting

### PR DESCRIPTION
**Pull request by chrisvanpatten**
## Description
When preloading requests via the preloading middleware, the middleware changes the shape of the returned response for `GET` requests based on the value of the `parse` flag. However, it does not do the same for `OPTIONS` requests, which are always returned in their unparsed form (with body and header properties).

This behavior appears to originate with the work around verifying media upload capabilities in #4155.

As far as I can tell, Gutenberg/Core only makes OPTIONS requests with `parse: false` as the requests are looking for header data, so this change should not impact core code. (It already receives the response shape it expects to get, and will continue to do so.)

## How has this been tested?
Added unit tests.

## Types of changes
This is technically a breaking change, if users implemented preloading expecting the incorrect response shape. I know there are larger discussions happening around issues with preloading, so it may make sense to punt this until those conversations are resolved.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by TimothyBJacobs** [comment]
Another inconsistency that would be nice to resolve is that unlike `GET` requests, if `parse` is `false`, we don't actually return a proper `fetch` `Response` object, but just a plain object.

**Comment by chrisvanpatten** [comment]
If we're comfortable breaking backcompat here and are overall happy with the approach, I'd be totally happy to implement that @TimothyBJacobs

**Comment by chrisvanpatten** [comment]
I have a version implemented locally that changes the response to use a true `window.Response` however it introduces some testing complexity because we'll have to mock it. So for now I'm not pushing that change, as I don't want to go down that path if we ultimately don't want to make the change anyway 😅

**Review by gziolo — APPROVED** [review]
It seems to be a good idea to align API for different methods of requests. Should we also include a note in the changelog of the package so people were aware of the new option added? Is there any higher level documentation that needs to be updated as well?

File: packages/api-fetch/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Breaking changes
+
+- `OPTIONS` requests which are handled by the preloading middleware are no longer resolved as unparsed responses unless you explicitly set `parse: false`, for consistency with other request methods. If you expect an unparsed response, add `{ parse: false }` to your request options to preserve the previous behavior.
```
Comment (gziolo) [inline packages/api-fetch/CHANGELOG.md:7]: Does it impact any existing REST API calls in Gutenberg?

Comment (chrisvanpatten) [reply packages/api-fetch/CHANGELOG.md:7]: It does not impact any existing calls in the Gutenberg codebase — I think the reason this wasn't noticed previously is that all the core requests already explicitly set `{ parse: false }`… it just wasn't actually _doing_ anything!

Comment (gziolo) [reply packages/api-fetch/CHANGELOG.md:7]: I can confirm it now that you mentioned it :)

**Review by gziolo — APPROVED** [review]

**Comment by chrisvanpatten** [comment]
@gziolo Thanks for merging — the performance tests were dragging on and I lost track of my open tab 😁

**Comment by gziolo** [comment]
Yes, it takes over 40 minutes, there is even an issue that prompts some action 😄

**Pull request merged**